src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx

```diff
@@ -0,0 +1,116 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { useContext } from 'react';
+import { jsx } from '@emotion/react';
+import path from 'ramda/src/path';
+import isEmpty from 'ramda/src/isEmpty';
+import useViewTracker from '#hooks/useViewTracker';
+import { ServiceContext } from '#app/contexts/ServiceContext';
+import PromoItem from '#app/legacy/components/OptimoPromos/PromoItem/index.styles';
+import PromoList from '#app/legacy/components/OptimoPromos/PromoList';
+import SectionLabel from '#psammead/psammead-section-label/src';
+import generatePromoId from '../generatePromoId';
+import LatestMediaItem from './LatestMediaItem';
+import styles from './index.styles';
+import {
+  LatestMedia,
+  TrackingBlock,
+  EventTrackingData,
+} from './LatestMediaTypes';
+
+const renderLatestMediaList = (
+  item: LatestMedia,
+  index: number,
+  eventTrackingData: EventTrackingData,
+  viewRef: () => Promise<void>,
+) => {
+  const ariaLabelledBy = generatePromoId({
+    sectionType: 'latest-media',
+    assetUri: null,
+    canonicalUrl: item.link,
+    uri: null,
+    contentType: item.type,
+    index,
+  });
+
+  return (
+    <PromoItem key={index} css={styles.LatestMediaPromoBorderAndWidth}>
+      <LatestMediaItem
+        item={item}
+        ariaLabelledBy={ariaLabelledBy}
+        ref={viewRef}
+        eventTrackingData={eventTrackingData}
+      />
+    </PromoItem>
+  );
+};
+
+const LatestMediaSection = ({ content }: { content: LatestMedia[] | null }) => {
+  const { service, dir, translations, script } = useContext(ServiceContext);
+
+  const eventTrackingData = {
+    block: {
+      componentName: 'latest-media',
+    },
+  };
+  const eventTrackingDataSend = path<TrackingBlock>(
+    ['block'],
+    eventTrackingData,
+  );
+  const viewRef = useViewTracker(eventTrackingDataSend);
+  const LABEL_ID = 'latest-media-heading';
+
+  if (!content || isEmpty(content)) return null;
+  const hasSingleContent = content.length === 1;
+  const singleItem = content[0];
+  const ariaLabelledBy = generatePromoId({
+    sectionType: 'latest-media',
+    assetUri: null,
+    canonicalUrl: singleItem.link,
+    uri: null,
+    contentType: singleItem.type,
+    index: 0,
+  });
```

Comment:
I am seeing a wee bit of duplication between this and within `renderLatestMediaList` above. Wondering if there might be a nicer way of consolidating this? i.e. instead of `renderLatestMediaList` could it just be `renderLatestMedia` and you pass a single latest media item to it?

Then you could reuse it in both places?

Alternatively, my preference (but maybe get a second opinion on this) would be to invoke the component code directly rather than abstracting it within a function - makes it much easier to read and understand what a component is doing. 

Will demonstrate 👇

Reply:
Fully agree with this. However, this method is used in other places [here](https://github.com/bbc/simorgh/blob/6b4e1ca6e949d26702e26a435a5acab655e4adef/src/app/pages/ArticlePage/PagePromoSections/RelatedContentSection/index.jsx#L33) and [here](https://github.com/bbc/simorgh/blob/6b4e1ca6e949d26702e26a435a5acab655e4adef/src/app/pages/ArticlePage/PagePromoSections/TopStoriesSection/index.jsx#L19). So maybe we scope that work in a separate ticket to tidy up this logic to not use a render function? What do you think?